### PR TITLE
Feature 2894 case archiving with batches

### PR DESCRIPTION
**Pull request by StefanKock**
Batching, JTA Transactions and logging is implemented. Performance is documented in #2894.

### CODE REVIEW

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/DateHelper.java [hunk 1]
```diff
@@ -1004,4 +1004,48 @@ public static Timestamp toTimestampUpper(Date date) {
 		timestamp.setNanos(timestamp.getNanos() + 999999);
 		return timestamp;
 	}
+
+	/**
+	 * Anzahl Millisekunden pro Sekunde.
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/utils/DateHelper.java:1009]: Translate to English; Also I'm not sure whether "Millies" is a common abbreviation in English, I'd prefer to use "Milliseconds" everywhere

File: sormas-backend/src/main/java/de/symeda/sormas/backend/util/ModelConstants.java [hunk 2]
```diff
@@ -23,4 +23,9 @@
 	String PERSISTENCE_UNIT_NAME_AUDITLOG = "auditlogPU";
 	String HINT_HIBERNATE_READ_ONLY = "org.hibernate.readOnly";
 
+	/**
+	 * Hard limit how much parameter can be allowed in a query.<br />
```
Comment (MateStrysewske) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/util/ModelConstants.java:27]: how much parameter => how many parameters
to big => too big
SQL-Statement => SQL statement

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/DateHelper.java (hunk 1 above)
Comment (StefanKock) [reply sormas-api/src/main/java/de/symeda/sormas/api/utils/DateHelper.java]: fixed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/util/ModelConstants.java (hunk 2 above)
Comment (StefanKock) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/util/ModelConstants.java:27]: fixed

**Pull request merged**